driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -504,14 +526,17 @@ public CodecRegistry register(Iterable<? extends TypeCodec<?>> codecs) {
 
     @SuppressWarnings("unchecked")
     private <T> TypeCodec<T> findCodec(DataType cqlType, TypeToken<T> javaType) {
-        checkNotNull(cqlType, "Parameter cqlType cannot be null");
-        logger.trace("Looking for codec [{} <-> {}]", cqlType, javaType == null ? "ANY" : javaType);
+        checkArgument(cqlType != null || javaType != null, "Parameter cqlType and javaType cannot be both null");
+        logger.trace("Looking for codec [{} <-> {}]", cqlType == null? "ANY" : cqlType, javaType == null ? "ANY" : javaType);
         for (TypeCodec<?> codec : codecs) {
-            if (codec.accepts(cqlType) && (javaType == null || codec.accepts(javaType))) {
+            if ((cqlType == null || codec.accepts(cqlType)) && (javaType == null || codec.accepts(javaType))) {
                 logger.trace("Codec found: {}", codec);
                 return (TypeCodec<T>) codec;
             }
         }
+        if (cqlType == null) {
```

Comment:
You are returning too soon; this will prevent the registry to create a codec for, say, `[list<int> <-> List<Foo>]` given an existing codec `[int <-> Foo]`.